Ginger/Ginger/ApplicationModelsLib/ModelParams Pages/ModelOptionalValuesPage.xaml.cs

```diff
@@ -36,41 +37,53 @@ namespace GingerWPF.ApplicationModelsLib.APIModelWizard
     public partial class ModelOptionalValuesPage : Page
     {
         GenericWindow mWin;
-        AppModelParameter mAMDP;
+        ObservableList<OptionalValue> mOptionalVal;
+        object mParentObject;
         bool editWasDone = false;
         bool mSelectionModePage;
 
-        public ModelOptionalValuesPage(AppModelParameter AMDP, bool selectionModePage = false)
+        public ModelOptionalValuesPage(object parentObject, string elementName, bool selectionModePage = false)
         {
```

Comment:
1- i prefer it to be more generic, change the constuctor to be:
ModelOptionalValuesPage(object parentObject, ObservablList<OptionalValue> optionalValues, string optionalValuesFieldName, string elementName, bool selectionModePage = false)
so "optionalValues" will be used as name and "parentObject" togather with "optionalValuesFieldName" will be used for "OnPropertyChanged" you need in later code in this class
2- why you sending HTMLElementInfo and not ElementInfo? this code should work for future POM's which will come from Windows/PB/Java in generic way

Reply:
Implemented changes